src/ast.ts

```diff
@@ -24,7 +24,8 @@ interface ASTNode {
   range?: vscodelc.Range;
 }
 const ASTRequestType =
-    new vscodelc.RequestType<ASTParams, ASTNode|null, void>('textDocument/ast');
+    new vscodelc.RequestType<ASTParams, ASTNode|undefined, void>(
```

Comment:
I don't understand this one, the JSON-RPC response may literally be `null`

Reply:
What I mean is, I think this type param corresponds to the actual JSON-RPC data sent by clangd, which definitely can be `null` (there's no `undefined` in JSON!).

So I think ASTRequestType needs to use `null` for correctness, but it's fine to convert that to `undefined` immediately after calling `sendRequest`.

Reply:
I was not aware of that. Then I agree, this should be changed back to `null` for the RequestType.